api/web_conn.go

```diff
@@ -118,6 +118,10 @@ func (c *WebConn) InvalidateCache() {
 	c.hasPermissionsToTeam = make(map[string]bool)
 }
 
+func (c *WebConn) InvalidateCacheForChannel(channelId string) {
```

Comment:
It needs to invalidate the cache of that channel for every user inside that channel, so I could reuse some more code by clearing that caches for each user totally but that seems wasteful